### PR TITLE
BMR20 Navigation

### PR DESCRIPTION
**Pull request by kamdaou-amalitech**
Added navigations
Renamed functions to match navigation flow
Removed unused codes
Testing can be done by navigating the app

### CODE REVIEW

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt [hunk 1]
```diff
@@ -3,30 +3,194 @@ package com.amalitech.bookmeetingroom
...
+}
+
+@Composable
+inline fun <reified T : ViewModel> NavBackStackEntry.sharedViewModel(navController: NavController): T {
+    val navGraphRoute = destination.parent?.route ?: return koinViewModel()
+    val parentEntry = remember(this) {
+        navController.getBackStackEntry(navGraphRoute)
     }
+    return koinViewModel(viewModelStoreOwner = parentEntry)
 }
```
Comment (esthcarelle) [inline app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt:194]: Can you create another file named AppNavigation? and extract the navigation composable to that class?

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt [hunk 2]
```diff
@@ -3,30 +3,194 @@ package com.amalitech.bookmeetingroom
...
+                        popUpTo(
+                            BottomNavItem.Home.route
+                        ) {
+                            saveState = true
+                        }
+                        launchSingleTop = true
+                        restoreState = true
+                    }
+                })
+            }
```
Comment (esthcarelle) [inline app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt:73]: This Scaffold should be in another composable, you should create another file for the NavHost

**Review by esthcarelle — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt [hunk 3]
```diff
@@ -3,30 +3,194 @@ package com.amalitech.bookmeetingroom
...
         setContent {
             BookMeetingRoomTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting(name = "Android")
-                }
+                val shouldShowOnboarding = sharedPref.loadShouldShowOnboarding()
```
Comment (bensek) [inline app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt:47]: Create a MainViewModel tied to the MainActivity that will hold state for the navigation and global information that can be shared across the entire application. Inject the sharedpreferences from there and expose a state called showOnboarding.

**Review by bensek — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt (hunk 3 above)
Comment (kamdaou-amalitech) [reply app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt]: The ViewModel is created

**Review by bensek — APPROVED** [review]

**Comment by esthcarelle** [comment]
Conflicts

**Pull request merged**